### PR TITLE
feat(cb2-5367): send api events from dynamics to eventbridge

### PR DESCRIPTION
**Pull request by aaron-downing**
## Send API Gateway events from Dynamics CE to Eventbridge

_**Not to be merged until tested in CB2-5367 environment, just wanting reviews**_

- New lambda (`vt-booking-dynamics-init`) to ingest requests made by Dynamics CE which contain a JSON payload  of DBA test bookings (in the form of an array of objects) within the request body.
- These are then sent onto an EventBridge to then be consumed by a subsequent lambda (`vt-booking`) that inserts them into the VEHICLE_BOOKING table

Related issue: [CB2-5367](https://dvsa.atlassian.net/browse/CB2-5236)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/services/eventbridge.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+): Promise<ISendResponse> => {
+  const sendResponse: ISendResponse = {
+    SuccessCount: 0,
+    FailCount: 0,
+  };
+  
+  logger.info('sendBooking starting');
+
+  for (const lineItem of lineItems) {
+    // eslint-disable-next-line security/detect-object-injection
```
Comment (Sul-Aga-kainos) [inline src/services/eventbridge.ts]: do we need these eslint disable statements? I have removed them and ran lint and didn't get any errors

File: src/services/eventbridge.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    logger.info(
+      `sending test booking with test date ${lineItem.testDate} for vehicle ${lineItem.vrm} at test station ${lineItem.pNumber} to eventbridge...`,
+    );
+
+    try {
+      const entry: IEventEntry = {
+        Source: config.aws.eventBusSource,
+        // eslint-disable-next-line security/detect-object-injection
```
Comment (Sul-Aga-kainos) [inline src/services/eventbridge.ts]: do we need these eslint disable statements? I have removed them and ran lint and didn't get any errors

File: src/services/eventbridge.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      };
+
+      const params: IEntries = {
+        Entries: [entry],
+      };
+
+      logger.debug(
+        `test booking event about to be sent: ${JSON.stringify(params)}`,
+      );
+      // eslint-disable-next-line no-await-in-loop
```
Comment (Sul-Aga-kainos) [inline src/services/eventbridge.ts]: same as above

File: src/services/eventbridge.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      await eventbridge.putEvents(params).promise();
+      sendResponse.SuccessCount++;
+    } catch (error) {
+      logger.error('', error);
```
Comment (Sul-Aga-kainos) [inline src/services/eventbridge.ts:45]: Worth adding some details to this error like an id or something `logger.error('Line item with id ${} fail', error);`

**Review by ghost — DISMISSED** [review]

**Review by naathanbrown — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]
Good refactoring on the test there Aaron

**Review by naathanbrown — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Review by naathanbrown — APPROVED** [review]

**Pull request merged**